compatibility/bazel_tools/testing.bzl

```diff
@@ -842,6 +843,15 @@ def sdk_platform_test(sdk_version, platform_version):
     if versions.is_at_least(kv_dedup_version, sdk_version) and versions.is_at_least(kv_dedup_version, platform_version):
         extra_sandbox_next_exclusions += ["--exclude=CommandDeduplicationIT", "--additional=KVCommandDeduplicationIT"]
 
+    # Error codes are enabled by default after 1.18.0-snapshot.20211117.8399.0.a05a40ae.
+    # Before this SDK version, ledger-api-test-tool cannot correctly assert the self-service error codes.
+    # For this reason, all platforms newer than 1.18.0-snapshot.20211117.8399.0.a05a40ae will run against
+    # old ledger-api-test-tools in compatibility mode (i.e. `--grpc-status-codes-compatibility-mode`)
+    error_codes_version_enabled_by_default = "1.18.0-snapshot.20211117.8399.0.a05a40ae"
+    if versions.is_at_most(error_codes_version_enabled_by_default, sdk_version):
+        extra_sandbox_next_args += ["--grpc-status-codes-compatibility-mode"]
+        extra_sandbox_classic_args += ["--grpc-status-codes-compatibility-mode"]
```

Comment:
@moritzkiefer-da @garyverhaegen-da @mziolekda

Reply:
Tested the compatibility tests locally with a sparser sub-set of versions.

Reply:
queued a full run at https://dev.azure.com/digitalasset/daml/_build/results?buildId=95477&view=results. Change looks very reasonable.

Reply:
Failures in some 1.18 snapshots
```
### FAILURES

KVCommandDeduplicationIT
- [KVCommandDeduplicationIT:KVCommandDeduplicationSimpleDeduplicationMixedClients] Deduplicate commands within the deduplication time window using the command client and the command submission client ... Unexpected failure (StatusRuntimeException)
  ALREADY_EXISTS: Duplicate commands
- [KVCommandDeduplicationIT:KVCommandDeduplicationSimpleDeduplicationBasic] Deduplicate commands within the deduplication duration window ... Assertion failed at line 427
  assertion failed: Expecting completion with status code OK but completion has status Some(Status(6,Duplicate commands,Vector(Any(type.googleapis.com/google.rpc.ErrorInfo,<ByteString@d03020 size=26 contents="\032\030\n\017definite_answer\022\005false">,UnknownFieldSet(Map()))),UnknownFieldSet(Map())))
- [KVCommandDeduplicationIT:KVCommandDeduplicationSimpleDeduplicationCommandClient] Deduplicate commands within the deduplication time window using the command client ... Unexpected failure (StatusRuntimeException)
  ALREADY_EXISTS: Duplicate commands
```

These are failing on main and are not related to the change in the error codes. I would not block on it

Reply:
@nicu-da may have some insight into the above failures.

Reply:
Does this need to be adopted to the the new rendition of the flags name?
```extra_sandbox_classic_args += ["--grpc-status-codes-compatibility-mode"]```

Reply:
Adapted 903a904ba787196a87042a6342574c926f4b46c1